File: gcd/python/gcd.py
##################################################################################################
#Euclidean GCD Algorithm to calculate Greatest Common Divisor of 2 Numbers
#gcd (X, Y) where X & Y can be any integers, regardless of whether X > Y or the other way
##################################################################################################

def gcd (x, y):
	x, y = max (x, y), min (x, y);
	while (y > 0):
		x, y = y, x % y;
	return (x);
	
#Alternatively, use the recursive Approach
def gcd_recursive (x, y):
	x, y = max (x, y), min (x, y);
	return (x if y == 0 else gcd (y, x % y));

#sample call to gcd ()
print (gcd (44, 36));
print (gcd_recursive (44, 36));

File: gcd/C++/gcd.cpp
#include <iostream>
using namespace std;

int gcd (int x, int y) {
	if (x < y) {
		int temp (x);
		x = y;
		y = temp;
	}

	while (y > 0) {
		int temp (y);
		y = x % y;
		x = temp;
	}

	return (x);
}

int gcd_recursive (int x, int y) {
	if (x < y) {
                int temp (x);
                x = y;
                y = temp;
        }

	if (y == 0) { return (x); }
	return (gcd_recursive (y, x % y));
}

int main () {
	cout << gcd (100, 45) << endl;
	cout << gcd_recursive (44, 36) << endl;

	return (0);
}
